### PR TITLE
Migrate sonarqube-backend plugin to the new backend system

### PR DESCRIPTION
**Pull request by andym0457**
## Hey, I just made a Pull Request!

Migrate sonarqube-backend plugin to the new backend system to contribute to the issue  #18301 

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR, there are some fixes required and it would be great to add a test for the new `sonarqubePlugin`

**Comment by andym0457** [comment]
> Thanks for the PR, there are some fixes required and it would be great to add a test for the new `sonarqubePlugin`

Hi jhaals,
I have tried to address all the comments, if there's any mistakes let me know. I'm not sure how to test the new sonarqubePlugin as I was following the other backend plugins that have been merged and couldn't see tests that have been added

**Review by jhaals — CHANGES_REQUESTED** [review]
This looks better with the router now being wired up properly.

The diff for yarn.lock is however very large so this branch is either out of date or you might have forgot to run `yarn`

**Comment by andym0457** [comment]
> This looks better with the router now being wired up properly.
> 
> The diff for yarn.lock is however very large so this branch is either out of date or you might have forgot to run `yarn`

Hi @jhaals,

Think the issue with the yarn.lock should now be fixed? Is there any other comments in relation to this PR?

**Comment by jhaals** [comment]
There's still a change needed to yarn.lock but a fairly small one. You can see that the post validation step is currently failing because `yarn` produces as diff. So if you commit the lock file from running yarn. It should be a quite small change.

**Review by jhaals — CHANGES_REQUESTED** [review]
CI Is currently failing because there are no updated API reports, please make sure to run `yarn build:api-reports` and commit the api report.

File: plugins/sonarqube-backend/src/index.ts [hunk 1]
```diff
@@ -16,3 +16,4 @@
 
 export * from './service/router';
 export * from './service/sonarqubeInfoProvider';
+export { sonarqubePlugin } from './plugin';
```
Comment (jhaals) [inline plugins/sonarqube-backend/src/index.ts:19]:
```suggestion
export { sonarqubePlugin as default } from './plugin';
```

File: packages/backend-next/src/index.ts [hunk 2]
```diff
@@ -45,4 +46,7 @@ backend.add(import('@backstage/plugin-search-backend/alpha'));
 backend.add(import('@backstage/plugin-techdocs-backend/alpha'));
 backend.add(import('@backstage/plugin-todo-backend'));
 
+// Sonarqube
+backend.add(sonarqubePlugin());
```
Comment (jhaals) [inline packages/backend-next/src/index.ts:50]: This is not how plugins are imported any longer, if you look at the example above this should be 
```suggestion
backend.add(import('@backstage/plugin-sonarqube-backend');
```

File: packages/backend-next/src/index.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@
  */
 
 import { createBackend } from '@backstage/backend-defaults';
+import { sonarqubePlugin } from '@backstage/plugin-sonarqube-backend';
```
Comment (jhaals) [inline packages/backend-next/src/index.ts:18]:
```suggestion
```

**Review by jhaals — CHANGES_REQUESTED** [review]
CI Is currently failing because there are no updated API reports, please make sure to run `yarn build:api-reports` and commit the api report.

File: plugins/sonarqube-backend/README.md [hunk 4]
```diff
@@ -2,6 +2,21 @@
...
+The Sonarqube backend plugin has support for the [new backend system](https://backstage.io/docs/backend-system/), here's how you can set that up:
+
+In your `packages/backend/src/index.ts` make the following changes:
+
+```diff
+  import { createBackend } from '@backstage/backend-defaults';
++ import { sonarqubePlugin } from '@backstage/plugin-sonarqube-backend';
+  const backend = createBackend();
+  // ... other feature additions
++ backend.add(sonarqubePlugin());
```
Comment (jhaals) [inline plugins/sonarqube-backend/README.md:16]:
```suggestion
+ backend.add(import('@backstage/plugin-sonarqube-backend');
```

File: plugins/sonarqube-backend/README.md [hunk 5]
```diff
@@ -2,6 +2,21 @@
...
 
+## New Backend System
+
+The Sonarqube backend plugin has support for the [new backend system](https://backstage.io/docs/backend-system/), here's how you can set that up:
+
+In your `packages/backend/src/index.ts` make the following changes:
+
+```diff
+  import { createBackend } from '@backstage/backend-defaults';
++ import { sonarqubePlugin } from '@backstage/plugin-sonarqube-backend';
```
Comment (jhaals) [inline plugins/sonarqube-backend/README.md:13]:
```suggestion
```

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks! Very small change left and then this is good to go 👍

**Review by jhaals — APPROVED** [review]
Thank you 👏

**Pull request merged**